### ISSUE TITLE
[pickers] Allows community to add new translations  

### ISSUE THREAD

**Issue by alexarnaud** [mui/mui-x#4500]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

On the DatePicker component, there are labels like "Previous month", used to provide a label for the low-vision and blind users using screen reader (accessibility). They are always display in English even if the local is for example French.

MUI provides natively no translation for those label and it seems there is no native way to contribute to translate them.

We can translate them on our side but it'd be better to provide them on the repo to benefit to all.

### Expected behavior 🤔

MUI would provide native translation for accessibility labels or provide a way for contributor to help translating them.

### Steps to reproduce 🕹

Steps:
1. Open a non-English page with a DatePicker in it.
2. Open the date picker calendar, check it is in your locale.
3. Inspect the "previous month" button, check if the string, attribute aria-label on the button, is in your locale.

Result:
It stays on English locale.

Expected result:
It should be in the user's locale.

### Context 🔦

When you're a low-vision or a blind person using screen reader or a motor disabled person using speech recognition, you rely on label. For screen reader users, it is to understand the meaning of the button, and for speech recognition users, it's to indicate to your tool where to click.

### Your environment 🌎

* Microsoft Windows 10.0.19041.1645
* Microsoft Edge 100.0.1185.39

**Comment by alexarnaud** [comment]
After searching again, I found https://github.com/mui/material-ui/issues/30998 on Material UI side, not sure in which repository this issue should go. As I understand, DatePicker component is here.

**Comment by alexfauquette** [comment]
Thanks for reporting this issue, we should replace all the hard encoded text to provide localization support customize `aria-label` and other helpers

**Comment by alexfauquette** [comment]
@alexarnaud After some investigation, you can override this specific level using the prop `leftArrowButtonText` and `rightArrowButtonText`which default values are 'Previous month' and 'Next month'.

I keep this issue open until we have a clear doc about how to support localization

**Comment by alexarnaud** [comment]
Thanks a lot for your feedback.

We had override it locally yet.

The main problem here seems to be we have no way to contribute back the translation in the repo?

**Comment by flaviendelangle** [comment]
We should indeed set a translation structure like the DataGrid has (and most of the other MUI components)

**Comment by subhamsood** [comment]
Do we have a similar prop for Timepicker as well? Right now **leftArrowButtonText** & **rightArrowButtonText** doesn't seem to work with DesktopTimePicker.

**Comment by flaviendelangle** [comment]
`leftArrowButtonText` and `rightArrowButtonText` have been deprecated in favor of `localeText`
For the `TimePicker` you should be able to use the `openPreviousView` and `openNextView` translation keys

[Working example](https://codesandbox.io/s/date-and-time-pickers-forked-9jj4cz?file=/src/App.tsx).

In v6, you will be able to pass it directly to the picker.

**Comment by subhamsood** [comment]
Thank You @flaviendelangle. This helped me to change the openPreviousView and openNextView text.
